packages/chat-stateful-client/src/ResourceDownloadQueue.ts

```diff
@@ -188,7 +194,12 @@ export const fetchImageSource = async (
     clearTimeout(id);
     return response;
   }
-  const accessToken = await credential.getToken();
+
+  const accessToken = await authentication.credential.getToken();
+  if (!src.startsWith(authentication.endpoint)) {
```

Comment:
Use https://developer.mozilla.org/en-US/docs/Web/API/URL/hostname
 to check the host name and protocol type